src/auth/bearer/fake/fake.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2014, Treehouse Networks Ltd. New Zealand
```

Comment:
> the copyright in question is part of the BSD license

Since the "copyright in question" is fully retained in CREDITS, we do not need it in this source file to satisfy the BSD license. The BSD license does not say "you must retain this notice in a file called foo/bar.cc".

> This is not newly minted code. ... its not my name there in the statement, and the year is not 2017.

I do not see the relevance of those facts. This is new code from the Squid Project p.o.v. (i.e., it did not exist in the official repository) and that is the only p.o.v. that matters here.

> My options here under the BSD License are to go in with the license + copyright as it was supplied with. Or to drop the helper entirely and hope something else gets written to fill the gap.

You have already accomplished the former (in the CREDITS file). No need to duplicate that info in this source file that may be co-authored by other contributors as time goes on.

> All the helpers and code that were written independently from Squid retain whatever their license and copyrights were under that umbrella Squid boilerplate.

I do not remember us agreed on such blanked exclusions. IMO, it does not matter whether the code was written for Squid or not (there is non-helper code in Squid that was not written for Squid and that does not have extra copyrights embedded into it). What matters is

- whether Squid contributors are likely to modify the code
- whether we can remove (or avoid adding) a custom copyright statement from (to) a particular source file.

Most of the leftover copyrights were grandfathered in because we did not have the energy to deal with them, not because something makes them a good idea that we should propagate.

> if you were to write a library (lets say paxxer) and place your own copyright statement on it, then either of us merged it wholesale into Squid that copyright would remain.

FWIW, I have removed or not added such statements from library-like code that Factory has contributed to Squid from other projects throughout the years. That does not prove me right, but at least I can say that I am eating my dog food (and intend to continue to do so).

> any modifications done afterwards would fall under the policy and not get additional copyright statements from the modifiers.

Creating the sad choice I have described earlier... When a custom copyright statement is placed close to the top of a source file, it looks like it applies to the whole file, and that send the wrong/misleading message to the future contributors and readers. The statement lives in the past while the code evolves. It is the wrong place to store that statement.

BTW, I do not care about the statement specifics (names, years, etc.). Any custom statement would create exactly the same set of problems.